cwms-data-api/src/main/java/cwms/cda/api/ForecastInstanceController.java

```diff
@@ -0,0 +1,209 @@
+                            + "implemented")
+            },
+            path = "/forecast-instance/all",
+            method = HttpMethod.GET,
+            tags = TAG
+    )
+    @Override
+    public void getAll(@NotNull Context ctx) {
+        try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+            throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        }
+    }
+
+    @OpenApi(
+            description = "Used to get all forecast instances for a given forecast spec",
+            queryParams = {
+                    @OpenApiParam(name = FORECAST_DATE_TIME, required = true, description = "Specifies the " +
+                            "forecast date time of the forecast instance to be retrieved."),
+                    @OpenApiParam(name = ISSUE_DATE_TIME, required = true, description = "Specifies the " +
+                            "issue date time of the forecast instance to be retrieved."),
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the " +
+                            "owning office of the forecast spec whose forecast instance is to be " +
+                            "included in the response."),
+                    @OpenApiParam(name = SPEC_ID, required = true, description = "Specifies the " +
+                            "spec id of the forecast spec whose forecast instance data is to be " +
+                            "included in the response."),
+                    @OpenApiParam(name = LOCATION, required = true, description = "Specifies the " +
+                            "location of the forecast spec whose forecast instance data to be included " +
+                            "in the response."),
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200,
+                            description = "A list of elements of the data set you've selected.",
+                            content = {
+                                    @OpenApiContent(from = ForecastInstance.class, type = Formats.JSONV2)}),
+                    @OpenApiResponse(status = STATUS_400, description = "Invalid parameter combination"),
+                    @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                            + "parameters did not find a forecast instance."),
+                    @OpenApiResponse(status = STATUS_501, description = "Requested format is not "
+                            + "implemented")
+            },
+            path = "/forecast-instance",
+            method = HttpMethod.GET,
+            tags = TAG
+    )
+    @Override
+    public void getOne(@NotNull Context ctx, @NotNull String id) {
+        try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+            throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+        }
+    }
+
+    @OpenApi(
+            description = "Update a forecast instance with provided values",
```

Comment:
Update would modify the key/value pair pieces and could replace the forecast file.

Reply:
maybe it would also be able to change the forecast date?

Reply:
what do you mean by key value pairs?

Reply:
Sorry, should have called that "forecast info", which is turning from a table to a column

Reply:
done